From 324eab1b148df1e8330b2dbbd114ab696c163e46 Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@theqtcompany.com>
Date: Mon, 11 May 2015 13:47:06 +0300
Subject: [PATCH 4/4] Generate usable qmake_extras.gypi

Running qmake recursively does not take .depends into account, so
the SUBDIRS need to be in correct order so that qmake_extras.gypi
is used for the ninja files. Do no try to write compiler tools as
it will break the build.

Task-number: QTBUG-45706
Upstream-Status: Pending
---
 src/core/core.pro                 | 7 ++++---
 src/core/gyp_configure_host.pro   | 3 ---
 src/core/gyp_configure_target.pro | 3 ---
 3 files changed, 4 insertions(+), 9 deletions(-)

diff --git a/src/core/core.pro b/src/core/core.pro
index cf00f39..747b711 100644
--- a/src/core/core.pro
+++ b/src/core/core.pro
@@ -12,9 +12,7 @@ gyp_run.depends = core_gyp_generator
 core_module.file = core_module.pro
 core_module.depends = gyp_run
 
-SUBDIRS += core_gyp_generator \
-           gyp_run \
-           core_module
+SUBDIRS += core_gyp_generator
 
 !win32 {
     # gyp_configure_host.pro and gyp_configure_target.pro are phony pro files that
@@ -28,3 +26,6 @@ SUBDIRS += core_gyp_generator \
     gyp_run.depends += gyp_configure_host gyp_configure_target
     SUBDIRS += gyp_configure_host gyp_configure_target
 }
+
+SUBDIRS += gyp_run \
+           core_module
diff --git a/src/core/gyp_configure_host.pro b/src/core/gyp_configure_host.pro
index eb94cb8..859bcbf 100644
--- a/src/core/gyp_configure_host.pro
+++ b/src/core/gyp_configure_host.pro
@@ -6,9 +6,6 @@ option(host_build)
 
 GYPI_CONTENTS = "{" \
                 "  'make_global_settings': [" \
-                "    ['CC.host', '$$which($$QMAKE_CC)']," \
-                "    ['CXX.host', '$$which($$QMAKE_CXX)']," \
-                "    ['LD.host', '$$which($$QMAKE_LINK)'],"
 
 GYPI_FILE = $$OUT_PWD/qmake_extras.gypi
 !build_pass {
diff --git a/src/core/gyp_configure_target.pro b/src/core/gyp_configure_target.pro
index 2a59966..317d24a 100644
--- a/src/core/gyp_configure_target.pro
+++ b/src/core/gyp_configure_target.pro
@@ -3,9 +3,6 @@ TEMPLATE = aux
 
 TOOLCHAIN_INCLUDES = $${QMAKE_INCDIR_EGL} $${INCLUDEPATH} $${QMAKE_INCDIR}
 
-GYPI_CONTENTS += "    ['CC', '$$which($$QMAKE_CC)']," \
-                 "    ['CXX', '$$which($$QMAKE_CXX)']," \
-                 "    ['LD', '$$which($$QMAKE_LINK)'],"
 GYPI_CONTENTS += "  ]," \
                  "  'target_defaults': {" \
                  "    'target_conditions': [" \
-- 
2.6.1

